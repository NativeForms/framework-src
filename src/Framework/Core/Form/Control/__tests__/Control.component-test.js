import React from 'react';
import { shallow } from 'enzyme';
import ControlComponent from '../Control.component';

describe('components <ControlComponent />', () => {
  it('renders label button component correctly', () => {
    const props = {
      type: 'button',
      attributes: {
        label: 'button test'
      }
    };
    const wrapper = shallow(
      <ControlComponent {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders label link component correctly', () => {
    const props = {
      type: 'link',
      attributes: {
        label: 'link test',
        route: 'http://www.google.com'
      }
    };
    const wrapper = shallow(
      <ControlComponent {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
