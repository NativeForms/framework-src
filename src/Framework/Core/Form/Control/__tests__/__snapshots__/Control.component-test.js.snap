// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <ControlComponent /> renders label button component correctly 1`] = `
<Styled(Item)
  style={
    Object {
      "borderBottomWidth": 0,
    }
  }
>
  <ButtonComponent
    label="button test"
  />
</Styled(Item)>
`;

exports[`components <ControlComponent /> renders label link component correctly 1`] = `
<Styled(Item)
  style={
    Object {
      "borderBottomWidth": 0,
    }
  }
>
  <LinkComponent
    label="link test"
    route="http://www.google.com"
  />
</Styled(Item)>
`;
